src/query/ee/src/storages/fuse/operations/vacuum_table.rs

```diff
@@ -375,6 +377,10 @@ pub async fn do_vacuum(
         Ok(Some(purge_files))
     } else {
         debug_assert!(dry_run_limit.is_none());
+        let retention = Duration::hours(ctx.get_settings().get_retention_period()? as i64);
+        // use min(now - get_retention_period(), retention_time) as gc orphan files retention time
+        // to protect files that generated by txn which has not been committed being gc.
+        let retention_time = std::cmp::min(chrono::Utc::now() - retention, retention_time);
```

Comment:
Should we use `base_snapshot.timestamp - retention` just like purge? @dantengsky

Reply:
local clock may bring troubles, consider a crazy fast lock which is used by `Utc::new()`…

for this PR, let’s temporarily ignore the above issue.

we need to address this later,  by marking the largest visible snapshot in the meta server (and in a “monotinically increasing way”, roughly.